onnxruntime/test/providers/cpu/tensor/transpose_test.cc

```diff
@@ -246,6 +246,39 @@ TEST(TransposeOpTest, ThreeDimSuffix) {
   TransposeTest(input_shape, input_vals, &perm, expected_shape, expected_vals, false);  //TensorRT: illegal error
 }
 
+TEST(TransposeOpTest, TransposeReshape) {
+  std::vector<int64_t> input_shape({1, 4, 2, 1, 3});
```

Comment:
Add comment explaining that the test is to hit the path in the Transpose implementation where we can reshape the input. May not be obvious to someone new to this code what 'transpose reshape' means.